git_machete/github.py

```diff
@@ -90,73 +97,131 @@ def _token_from_hub() -> Optional[str]:
+    token: Optional[str] = github_token()
+    request_body: Dict[str, List[str]] = {
+        'assignees': assignees
+    }
+    # Adding assignees is only available via the Issues API, not PRs API.
+    __fire_github_api_request('POST', f'/repos/{org}/{repo}/issues/{number}/assignees', token, request_body)
+
+
+def add_reviewers_to_pull_request(org: str, repo: str, number: int, reviewers: List[str]) -> None:
+    token: Optional[str] = github_token()
+    request_body: Dict[str, List[str]] = {
+        'reviewers': reviewers
+    }
+    __fire_github_api_request('POST', f'/repos/{org}/{repo}/pulls/{number}/requested_reviewers', token, request_body)
+
 
+def set_base_of_pull_request(org: str, repo: str, number: int, base: str) -> None:
     token: Optional[str] = github_token()
-    prs = fire_github_api_get_request(f'/repos/{org}/{repo}/pulls', token)
-    return [GitHubPullRequest(int(pr['number']), pr['user']['login'], pr['base']['ref'], pr['head']['ref']) for pr in prs]
+    request_body: Dict[str, str] = {'base': base}
+    __fire_github_api_request('PATCH', f'/repos/{org}/{repo}/pulls/{number}', token, request_body)
+
+
+def set_milestone_of_pull_request(org: str, repo: str, number: int, milestone: str) -> None:
+    token: Optional[str] = github_token()
+    request_body: Dict[str, str] = {'milestone': milestone}
+    # Setting milestone is only available via the Issues API, not PRs API.
+    __fire_github_api_request('PATCH', f'/repos/{org}/{repo}/issues/{number}', token, request_body)
+
+
+def derive_pull_request_by_head(org: str, repo: str, head: str) -> Optional[GitHubPullRequest]:
+    token: Optional[str] = github_token()
+    prs = __fire_github_api_request('GET', f'/repos/{org}/{repo}/pulls?head={org}:{head}', token)
+    if len(prs) >= 1:
+        return parse_pr_json(prs[0])
+    else:
+        return None
+
+
+def derive_pull_requests(org: str, repo: str) -> List[GitHubPullRequest]:
+    token: Optional[str] = github_token()
+    prs = __fire_github_api_request('GET', f'/repos/{org}/{repo}/pulls', token)
+    return list(map(parse_pr_json, prs))
 
 
 def derive_current_user_login() -> Optional[str]:
     token: Optional[str] = github_token()
     if not token:
         return None
-    user = fire_github_api_get_request('/user', token)
+    user = __fire_github_api_request('GET', '/user', token)
     return str(user['login'])  # str() to satisfy mypy
 
 
 def is_github_remote_url(url: str) -> bool:
-
     return any((re.match(pattern, url) for pattern in GITHUB_REMOTE_PATTERNS))
 
 
-def parse_github_remote_url(url: str) -> Optional[Tuple[str, str]]:
-
+def parse_github_remote_url(github_url: str) -> Tuple[str, str]:
     for pattern in GITHUB_REMOTE_PATTERNS:
-        match = re.match(pattern, url)
+        match = re.match(pattern, github_url)
         if match:
             return match.group(1), match.group(2)
-    return None
+    raise MacheteException(f'{github_url} is not a GitHub remote URL')
```

Comment:
Hmmm judging by how `parse_github_remote_url` is used currently... should really an exceptionbe thrown here? 🤔 what if a local repository has a few remotes, some of which are GitHub and some aren't?